### ISSUE TITLE
Shareing not working

### ISSUE THREAD

**Issue by hermann1514** [owncloud/documents#622]
When i will create a new Share, i can choose the User but then only the rotating circle is shown on the right side and in the middle of the scrren there is an errror "failed to share"

**Comment by rullzer** [comment]
Ah it seems the sharing from the documents app is still referencing the obsolete (and removed) internal private api endpoint. We had the same issue with the gallery. I'll see if it is easy to port the patch from @oparoz .

**Comment by oparoz** [comment]
Bear in mind that the upload and email buttons are not activated yet in Gallery. There is work currently done to make it happen.

**Comment by rullzer** [comment]
Mmmm crap. The documents app won't really work on my system. (at least the default install steps fail)... so this makes working on this issue a bit harder...

**Comment by rullzer** [comment]
Ah some fiddling made it work... let see..

**Comment by rullzer** [comment]
mmm so it seems that the path is not present easily from within the documents app...

**Comment by rullzer** [comment]
@oparoz am I correct in assuming that with modifying the `namespace` copying over your changes should not create much trouble?

**Comment by oparoz** [comment]
Correct, with the caveats mentioned above. As long as you have the path store in the share button, it should work.

**Comment by rullzer** [comment]
@oparoz cool! then I'll try to tackle the path issue and then just pull your file :) Thanks

**Comment by resoli** [comment]
I'm hit as well by this bug. No way to share a document link for collaborative editing.

**Comment by rullzer** [comment]
@resoli, as a work around share the file via the 'files' app.

**Comment by resoli** [comment]
Hello @rullzer; this is not an option, I need to share under collaborative editing ... I will wait for a fix.

**Comment by reuterbal** [comment]
Same problem here on OC 9.0.0:
- Installed master branch: works fine but sharing issue persists
- Applying #624 doesn't fix the issue for me, now the entire share button isn't working anymore

**Comment by reuterbal** [comment]
Sorry, not-working share button was probably due to the bug in Mail app owncloud/mail#1422.

When clicking the share button, it produces now the error message "Error while retrieving shares" and the associated request to
`/ocs/v2.php/apps/files_sharing/api/v1/shares?format=json&path=7563&shared_with_me=true`
fails with 404.

**Comment by oparoz** [comment]
@reuterbal - Have you disabled any of the sharing options in admin?

**Comment by reuterbal** [comment]
Not that I'm aware of. Specifically the Share API is enabled:
![screenshot from 2016-04-14 14 45 27](https://cloud.githubusercontent.com/assets/6384870/14528429/99c5117e-024f-11e6-99c6-4f82c3af35b7.png)

**Comment by oparoz** [comment]
Thanks. What do you get if you just paste the URL in your browser?

**Comment by reuterbal** [comment]
It asks me for authentication repeatedly, not accepting (correct) credentials.
`curl -u <user>:<pass> ...` provides me with a list of shares.

**Comment by oparoz** [comment]
Thanks, so that URL works. I wonder why it gives you a 404 when called from within documents

**Comment by reuterbal** [comment]
Sorry, mistake on my side with the curl-call. I did not encapsulate the URL in quotation marks, so not all GET parameters were sent.
```curl .../ocs/v2.php/apps/files_sharing/api/v1/shares?format=json&shared_with_me=true"```
works, however
```curl .../ocs/v2.php/apps/files_sharing/api/v1/shares?format=json&path=7563&shared_with_me=true"```
does not (note the additional path=7563)

**Comment by reuterbal** [comment]
(works not = 404)

**Comment by oparoz** [comment]
Thanks, that should help figure out what the problem is.

**Comment by Angristan** [comment]
Hello, I also have a issue with file sharing, but it does not show any error, it just loads
![screenshot_24-04-2016_gcivlph4](https://cloud.githubusercontent.com/assets/11699655/14785309/6196b9de-0afa-11e6-8ab0-f6d885b868a3.png)

**Comment by retide** [comment]
@Angristan: could be the same issue - I get an error message on Firefox and the spinning wheel on Safari. In any case sharing from the documents app currently does not work for me either.

**Comment by reuterbal** [comment]
Just upgraded PHP 5.5.9 to PHP 5.6.22 - this fixed login problems with LDAP-Users from Android Apps. However, sharing of documents does still not work.

**Comment by kidandcat** [comment]
+1
Some quick fix until it is resolved?

I have nothing logged, but it keep loading like in @Angristan pic.

**Comment by ObviouslyNot** [comment]
Is anything happening here soon? Same problem for me: Error message in firefox, endless spinning in chrome. Changes from the pull request didn't fix it.

In one case there was a link generated eventually, which did, however, not show the document, but an empty page with the owncloud logo. Could not replicate that though.

**Comment by olivierche** [comment]
Same problem with owncloud-files-9.1.0-1.1.noarch and  documents: 0.13.1 on CentOS Linux release 7.2.1511 (Core).

When i try to share a doc i got ""failed to share / Erreur lors de la mise en partage" in firefox, and in owncloud.log : {"reqId":"qdfsqfdqfqf","remoteAddr":"MYIP","app":"PHP","message":"Invalid argument supplied for foreach() at \/var\/www\/html\/owncloud\/core\/ajax\/share.php#342","level":3,"time":"2016-08-23T13:26:03+02:00","method":"GET","url":"\/index.php\/core\/ajax\/share.php?fetch=getShareWith&search=usernameofuseritrytosharedocwith&limit=200&itemType=file","user":"myldaplogin"}

Enabled:
  - activity: 2.3.2
  - comments: 0.3.0
  - dav: 0.2.5
  - documents: 0.13.1
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.10.0
  - files_texteditor: 2.1
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 15.0.0
  - notifications: 0.3.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.1
  - updatenotification: 0.2.1
  - user_ldap: 0.9.0
Disabled:
  - encryption
  - external
  - files_antivirus
  - files_external
  - user_external

**Comment by joker234** [comment]
same problem here with documents 0.13.1 and ownCloud 9.1.0 (from Debian owncloud stable repos). would appreciate soon fix of this issue :)

**Comment by olivierche** [comment]
Same problem after upgrade to ownCloud 9.1.1(.3  from Centos 7 ownCloud stable repos) and  documents: 0.13.1.
Is a fix planed for 9.1 branch ?

**Comment by kidandcat** [comment]
Nothing planed and no response in months, too much time waiting, next week
I will fork the app and take care of it, any help will be welcome :)

**Comment by tiefenau** [comment]
Maybe the bug has something to do with the encryption-module. I found the following message in my log after trying to share a file:

"Message":"Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you."

**Comment by kidandcat** [comment]
The bug is that the internal owncloud API changed from owncloud 8 to
owncloud 9, so now the methods used to generate the shared link doesn't
work anymore.

**Comment by PVince81** [comment]
> I will fork the app and take care of it, any help will be welcome :) 

@kidandcat Have you done so already ? Do you have a link ?

Note that the documents app has no maintainer currently so if someone is up to taking care of it and publishing to the app store (and future market), please let us know.

**Comment by kidandcat** [comment]
@PVince81 No, I'm focushing now in [CODE](https://www.collaboraoffice.com/code/)

**Comment by olivierche** [comment]
Hi,
@PVince81. Is there a new maintainer for the documents app that could work on this issue (for ownCloud 9.1.5 or 10) ?
It seems that there's a lot of users waiting for it to work again (since 2016).

Reading https://doc.owncloud.org/server/9.1/admin_manual/configuration_files/collaborative_documents_configuration.html one can think that "Documents application supports share and edit files in the browser, and then share them inside ownCloud or through a public link." but that's no more true.
I'm sorry i can't help (i'm only sysadmin ;) ) but **i'd like to know if there's a roadmap for this bug**. Without maintainer in the next days, i'll try to install a server with Collabora Online and install the "Collabora Online connector" apps but this is much more work than installing the official documents app !
Regards,
Olivier

**Comment by PVince81** [comment]
There is currently no maintainer for the app and no plans to release new versions.

If someone wants to step up and maintain and make releases for this app, feel free.

**Comment by Rudloff** [comment]
Can someone please sum up what needs to be changed in order to fix this bug?

**Comment by VicDeo** [comment]
@Rudloff  In short: Write a js code using a new sharing dialog.
